src/zarr/abc/store.py

```diff
@@ -22,8 +22,20 @@ def mode(self) -> OpenMode:
     @property
     def writeable(self) -> bool:
```

Comment:
instead of multiple properties that all interpret `self.mode` in different ways, perhaps we should just make `self.mode` more expressive? E.g., suppose `self.mode` was an instance of the following type:

```
WriteMode = {
    create: bool,
    overwrite :bool,
},
write: False | WriteMode
}
```

I'm sure someone can come up with something better. But I think we should consider this approach (making one property more expressive) before we add a lot of properties to the store.

Reply:
good idea!

Reply:
do you mean to remove the literals` ["r", "r+", "a", "w", "w-"] `completely? or just organize the properties better into `WriteMode`

Reply:
I think we could accept the literals as parameters in the constructor, but we should consider normalizing them to a more explicit data structure.